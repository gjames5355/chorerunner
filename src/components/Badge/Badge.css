.levelInfo {
  background: rgba(0, 0, 0, 0.7);
  border-radius: 10px;
  width: 95%;
  margin: 20px auto;
  justify-content: space-around;
}


.badge {
  width: 30%;
  padding: 10px;
  margin: 10px auto;
  height: auto;
  min-height: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
}


.level {
  width: 90%;
  padding: 5px;
  margin: 10px auto;
  height: auto;
  font-family: 'Press Start 2P', cursive;
  color: white;
  text-align: left;
}

.image img {
  padding-top: 25px;
  height: 200px;
}

.level p {
  font-family: 'Press Start 2P', cursive;
  text-align: left;
}

.level_totalScore {
  text-align: center;
}



@media screen and (min-width: 600px) {

  .levelInfo {
    background: rgba(0, 0, 0, 0.7);
    border-radius: 10px;
    width: 80%;
    margin: 20px auto;
    display: flex;
    justify-content: space-between;
  }


  .badge {
    width: 60%;
    padding: 10px;
    margin: 10px auto;
    height: auto;
    min-height: 100px;
    display: flex;
    align-items: left;
    justify-content: center;
  }

  .image img {
    padding-top: 25px;
    height: 250px;
  }


  .level {
    width: 40%;
    padding: 10px;
    margin: 10px auto;
    height: auto;
    min-height: 100px;
    font-family: 'Press Start 2P', cursive;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    color: white;
  }


  .level p {
    font-family: 'Press Start 2P', cursive;
    width: 90%;
    text-align: left;
  }
}

@media screen and (min-width: 850px) {
  .levelInfo {
    width: 60%;
    justify-content: space-evenly;
  }

  .badge {
    width: 50%;
    padding: 10px;
    margin: 10px auto;
    height: auto;
    min-height: 20px;
    display: flex;
    align-items: center;
    justify-content: center;
  }


  .level {
    width: 50%;
    padding: 5px;
    margin: 10px auto;
    height: auto;
    font-family: 'Press Start 2P', cursive;
    color: white;
    text-align: left;
  }

  .image img {
    padding-top: 25px;
    height: 200px;
  }

  .level p {
    font-family: 'Press Start 2P', cursive;
    text-align: left;
    /* margin: 0; */
  }

  .level_totalScore {
    text-align: center;
  }
}

@media screen and (min-width: 1050px) {
  .levelInfo {
    width: 700px;
    justify-content: space-evenly;
  }
}