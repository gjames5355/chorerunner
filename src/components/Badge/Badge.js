import React, { Component } from 'react';
import ApiService from '../../services/api-service';
import './Badge.css';
import images from '../../ImgAssets/index';

export default class Badge extends Component {
  state = {
    levelInfo: {},
  };

  componentDidMount() {
    ApiService.getBadge().then(res => {
      this.setState({ levelInfo: res });
    });
  }

  renderBadge() {
    const { levelInfo } = this.state;
    return (
      <section className='levelInfo'>
        <div className='badge'>
          <div className='image'>
            <img src={images[levelInfo.badge]} alt='Badge png' />
          </div>
        </div>
        <div className='level'>
          <div className='level_totalScore'>
            <p>{levelInfo.name}</p>
            <p>Level {levelInfo.level_id}</p>
            <p>Total EXP: {levelInfo.total_score}</p>
            <p>EXP needed to level up: {levelInfo.nextLevel}</p>
          </div>
        </div>
      </section>
    );
  }

  render() {
    if (this.state.levelInfo.badge) {
      return this.renderBadge();
    }
    const { levelInfo } = this.state;
    return <>{!!levelInfo.badge ? this.renderBadge() : <p> LOADING... </p>}</>;
  }
}
