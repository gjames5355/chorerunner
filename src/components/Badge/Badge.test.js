import React from 'react';
import ReactDOM from 'react-dom';
import Badge from './Badge';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Badge component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <Badge />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Badge />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('reads the components State', () => {
    const wrapper = shallow(<Badge />);
    expect(wrapper.state('levelInfo')).toEqual({});
  });
});
