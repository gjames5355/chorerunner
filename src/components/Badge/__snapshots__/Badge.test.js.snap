// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component testing renders UI as expected 1`] = `
<p>
   LOADING... 
</p>
`;
